src/main/kotlin/org/rust/lang/core/stubs/types/RustItemStubElementType.kt

```diff
@@ -0,0 +1,21 @@
+package org.rust.lang.core.stubs.types
+
+import com.intellij.psi.stubs.StubElement
+import com.intellij.psi.stubs.StubInputStream
+import com.intellij.psi.stubs.StubOutputStream
+import org.rust.lang.core.psi.RustItem
+import org.rust.lang.core.stubs.RustItemStub
+
+abstract class RustItemStubElementType<PsiT: RustItem>(debugName: String)
+    : RustStubElementType<RustItemStub, PsiT>(debugName) {
+
+    override fun serialize(stub: RustItemStub, dataStream: StubOutputStream) =
+        dataStream.writeName(stub.name)
+
+    override fun deserialize(dataStream: StubInputStream, parentStub: StubElement<*>?): RustItemStub =
+        RustItemStub(parentStub, this, dataStream.readName())
+
+    override fun createStub(psi: PsiT, parentStub: StubElement<*>?): RustItemStub =
+        RustItemStub(parentStub, this, psi.name ?: "")
```

Comment:
I think this `psi.name ?: ""` is wrong, but I don't know what the correct solution is.

I don't quite undestand the contract of `NamedStubBase`. It accepts a `null` name in [constructors](https://github.com/JetBrains/intellij-community/blob/master/platform/lang-api/src/com/intellij/psi/stubs/NamedStubBase.java#L29) and then fails at run time with NPE in [`getName`](https://github.com/JetBrains/intellij-community/blob/master/platform/lang-api/src/com/intellij/psi/stubs/NamedStubBase.java#L40). So I hacked around it by `?: ""`.